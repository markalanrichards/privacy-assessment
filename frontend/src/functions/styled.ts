import * as styledComponents from "styled-components";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>;

interface Color {
  primaryColor: string;
}

interface Size {
  large: string;
}

export interface IThemeInterface {
  color: Color;
  size: Size;
}

export const defaultTheme: IThemeInterface = {
  color: {
    primaryColor: "yellow"
  },
  size: {
    large: "10"
  }
};

export default styled;
export { css, injectGlobal, keyframes, ThemeProvider };
