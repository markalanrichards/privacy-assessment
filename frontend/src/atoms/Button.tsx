import * as React from "react";
import styled from "../functions/styled";

export interface AtomProps {
  url: string;
  title: string;
}

const DefaultButton = styled.button`
  color: ${props => props.theme.color.primaryColor};
  font-size: ${props => props.theme.size.large};
  border: none;
`;

export const Button: React.SFC<AtomProps> = ({ url, title }) => {
  return (
    <a href={url}>
      <DefaultButton type="button">{title}</DefaultButton>
    </a>
  );
};
