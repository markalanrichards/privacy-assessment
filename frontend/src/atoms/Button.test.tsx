import * as React from "react";
import { mount, shallow } from "enzyme";
import styled, { ThemeProvider } from "../functions/styled";
import { Button } from "./Button";

const defaultTheme = {
  color: {
    primaryColor: "yellow"
  },
  size: {
    large: "10"
  }
};

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    // @ts-ignore
    .getChildContext(); // not in enzyme types file GRRRR!!
  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  });
};

describe("Title component With Theme", () => {
  test("Normal", () => {
    const wrapper = shallowWithTheme(
      <Button title="title" url="https://test.com" />,
      defaultTheme
    );
    expect(wrapper).toMatchSnapshot();
  });
});
