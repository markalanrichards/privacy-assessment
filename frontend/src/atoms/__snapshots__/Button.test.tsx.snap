// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component With Theme Normal 1`] = `
<Component
  title="title"
  url="https://test.com"
>
  <a
    href="https://test.com"
  >
    <styled.button
      type="button"
    >
      <button
        className="sc-bdVaJa ktQjs"
        type="button"
      >
        title
      </button>
    </styled.button>
  </a>
</Component>
`;
